spring-pulsar/src/main/java/org/springframework/pulsar/aot/PulsarRuntimeHints.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.aot;
+
+import java.util.HashSet;
+import java.util.TreeMap;
+import java.util.stream.Stream;
+
+import org.apache.pulsar.client.admin.internal.OffloadProcessStatusImpl;
+import org.apache.pulsar.client.admin.internal.PulsarAdminBuilderImpl;
+import org.apache.pulsar.client.api.Authentication;
+import org.apache.pulsar.client.api.AuthenticationDataProvider;
+import org.apache.pulsar.client.impl.conf.ClientConfigurationData;
+import org.apache.pulsar.client.impl.conf.ConsumerConfigurationData;
+import org.apache.pulsar.client.impl.conf.ProducerConfigurationData;
+import org.apache.pulsar.client.util.SecretsSerializer;
+import org.apache.pulsar.common.protocol.Commands;
+import org.apache.pulsar.shade.io.netty.buffer.AbstractByteBufAllocator;
+import org.apache.pulsar.shade.io.netty.channel.socket.nio.NioDatagramChannel;
+import org.apache.pulsar.shade.io.netty.channel.socket.nio.NioSocketChannel;
+import org.apache.pulsar.shade.io.netty.util.ReferenceCountUtil;
+
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.ReflectionHints;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+import org.springframework.lang.Nullable;
+
+/**
+ * {@link RuntimeHintsRegistrar} for Spring for Apache Pulsar.
+ *
+ * @author Soby Chacko
+ */
+public class PulsarRuntimeHints implements RuntimeHintsRegistrar {
+
+	@Override
+	public void registerHints(RuntimeHints hints, @Nullable ClassLoader classLoader) {
+		ReflectionHints reflectionHints = hints.reflection();
+		Stream.of(HashSet.class, TreeMap.class, Authentication.class, AuthenticationDataProvider.class,
```

Comment:
Do. you mind adding a 1 liner comment about each section describing what/why etc? I think it may be priceless when we look back in here in like 6months and say "wuuuttt?" 😄